katsdpimageutils/primarybeamcorrection/primary_beam_correction.py

```diff
@@ -1,188 +1,219 @@
+    # The third and fourth argument is "origin" -- in this case we're declaring we
+    # have 0-based (Numpy-like) coordinates.
+    p2w = image_wcs.pixel_to_world(col, row, 0, 0)[0]
+    # A separation vector between phase centre and source positions in radians.
     separation_rad = p2w.separation(phase_center).rad
     return separation_rad
 
 
 def central_freq(path):
-    """
-    Function to get central frequency of each frequency plane.
-    Input : image header
-    Output : List of central frequency in MHz of each frequency plane.
+    """Determine central frequency of each frequency plane.
+    
+    Parameters
+    ----------
+    Input : str
+        FITS file
+    Output : numpy array
+        An array of central frequencie in MHz of each frequency plane.
     """
     images = read_fits(path)
     c_freq_plane = []
+    # NSPEC -  number of frequency planes.
     for i in range(1, images.header['NSPEC']+1):
+        # FREQ00X is the central frequency for each plane X.
         c_freq_plane.append(images.header['FREQ{0:04}'.format(i)])
     return np.array(c_freq_plane)
 
 
 def cosine_power_pattern(separation_rad, c_freq):
-    """
-    Power patterns for a given frequency based on the Cosine-squared power approximation
-    from Mauch et al. (2020).
-    Input: Radial separation array and a central frequency
+    """Compute Power patterns for a given frequency based on the
+    Cosine-squared power approximation from Mauch et al. (2020).
+    
+    Parameters
+    ----------
+    separation_rad: numpy array
+        Radial separation array
+    c_freq: numpy array
+        An array of central frequencies for each frequency plane
     """
     rho = separation_rad
     # Degrees to radians conversion
-    conv = np.pi / 180.
-    # converting arcminutes to radians
-    v_beam_rad = 89.5 / 60. * conv
-    h_beam_rad = 86.2 / 60. * conv
-    # Taking the Geometric mean for the veritical and horizontal cut through the beam.
-    vh_beam_mean = np.sqrt(v_beam_rad*h_beam_rad)
+    v_beam_rad = np.deg2rad(89.5 / 60.)
+    h_beam_rad = np.deg2rad(86.2 / 60.)
+    # Taking the Geometric mean for the vertical and horizontal cut through the beam.
+    vh_beam_mean = np.sqrt(v_beam_rad * h_beam_rad)
     flux_density = []
     for nu in c_freq:
+        # Converting GHz to Hz
         nu = nu/1.e9
-        theta_b = vh_beam_mean*(1./nu)
+        theta_b = vh_beam_mean / nu
         ratio = rho/theta_b
-        num = np.cos(1.189*np.pi*ratio)
-        dem = 1-4*(1.189*ratio)**2
-        a_b = (num/dem)**2
+        num = np.cos(1.189 * np.pi * ratio)
+        den = 1-4 * (1.189 * ratio)**2
+        a_b = (num/den)**2
         flux_density.append(a_b)
     return flux_density
 
 
 def beam_pattern(path):
-    """
-    Making beam pattern using the Cosine-squared power approximation from Mauch et al. (2020)
+    """Make beam pattern using the Cosine-squared power approximation
+    from Mauch et al. (2020).
```

Comment:
Again, make the first line a summary, followed by a blank line.

Reply:
Done.